userspace/libsinsp/container_engine/docker_common.cpp

```diff
@@ -89,17 +89,29 @@ bool docker::parse_docker(sinsp_container_manager* manager, sinsp_container_info
 
 	container->parse_healthcheck(config_obj["Healthcheck"]);
 
-	// Saving full healthcheck for container event parsing/writing
+	// Saving full object for container event parsing/writing
 	container->m_healthcheck_obj = config_obj["Healthcheck"];
 
+	// For k8s liveness/readiness probes, we need to fetch a
+	// specific label that contains the pod spec, which is
+	// stringified json
+	Json::Value pod_spec_obj = get_k8s_pod_spec(config_obj);
```

Comment:
we are now supporting containerd with k8s and in the process of adding cri-o support. is there a way to add this capability in a non-docker specific way? btw, i don't know if containerd or cri-o containers would have the `last_applied_configuration` in a label or would require a separate query to the CRI endpoint. if its not available from either source, would we need to get this from the k8s api server in the agent?

Reply:
I think what I have is pretty generic now, with a list of health probes objects with types. Assuming that the json is the same, it should just result in a call to ` sinsp_container_info::container_health_probe::add_health_probes` when getting the container metadata. If it's not the same kind of json, I'll add support for however it's represented.

Reply:
With the reorganization I just made, when we add support for cri we'll do this in the cri engine resolver. If it happens that it can re-use code from the docker engine resolver, I'll share methods between them.